Helm/depdraw/charts/k8sclient/templates/k8sclient-build.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: build.openshift.io/v1
+kind: BuildConfig
+metadata:
+  name: depdraw-{{ default .Release.Name "k8sclient" }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  output:
+    to:
+      kind: ImageStreamTag
+      name: {{ .Release.Name }}-k8sclient:latest
+      namespace: {{ .Release.Namespace }}
+  postCommit: {}
+  source:
+    git:
+      uri: {{ default "https://github.com/nemerna/DepDraw.git" .Values.ghRepo }}
```

Comment:
i see 2 issues
1) i forgot to change the repo from my fork to our official repo 
2) yes you are right I will make the chart more simple I will hardcode the things that are not really needed to be changed

Reply:
@nemerna @dmartinol 
If it's something that is not planned to be change, then it's not needed, and it can be defaulted in chart' values, as key in original values.yaml is the default anyway, and even if the key omitted from override/external  values.yaml, then it will take the default from charts' default values.yaml(like --set flag which overrides only certain keys' values and not affecting keys that weren't specified).
In essence , the only condition in which default will come in handy in that case:

1. You're using helm install with override/external values.yaml, or using the original values.yaml
and
2. you specified the key in there but with no value(cleared the value, for example)

That is,  you cleared the default values, or tell helm to intentionally override the default with blank value.
In this case , helm template engine will render the template with key without matching value(empty), if a default function is present in the template, then the value will be fetched from it in this case, otherwise, key will be with no values at all, and this could lead to some errors,  off course.

Reply:
i have used the variable ghRepo for letting people who for our repo still be able to use the chart

but removed the default function as the ghRepo variable is defaulted to our one

Reply:
Thanks! @zvigrinberg @dmartinol 

i believe now all good

but had 1 thing to mention, the build process of the k8sclient makes use of the docker file
which uses the base image ALPINE:3, and it reaches the anonymous pull limit from docker hub.

i will be happy for suggestions,
1) have the image in some quay that we own
2) arrange some docker-pull-secret
3) any other solution that you guys know/ used before

Reply:
@nemerna 
If you only need  ALPINE:3, and you don't anticipate to rely on updates to the base image(that is,  ALPINE:latest) then you can push this image into quay.io as public repository in your account.
If you need the latest(sometimes has side effects, but it's another story), then more recommended , is to arrange some image pull secret containing credentials for docker.io account, as part of the automation of the helm chart.
There are couple of ways to do so, the following is highly suggested:

[helm - Creating Image Pull Secrets](https://helm.sh/docs/howto/charts_tips_and_tricks/#creating-image-pull-secrets)
Just don't forget not to push into repo your credentials to docker.io, keep it only locally in your computer.
If you're interested in putting it in the repo, we can consider use Gnu-PG, SOPS and helm secrets plugin, and encrypt the sensitive data that creating image pull secret(but this can be performed later on).